### DISCUSSION TITLE
Is it a good practice to create a new hidden window as a child JavaScript process

### DISCUSSION THREAD

**Discussion by yf-yang** [tauri-apps/tauri#8174] [Q&A] [answered]
Say I want to have independent processes to execute untrusted error-prone third party JavaScript plugins (pretty similar to vscode extensionhost design). In electron, [UtilityProcess](https://electronjs.org/docs/latest/api/utility-process) is an option. What is the equivalent approach in tauri if those codes are JavaScript?

Here are some possible approaches:
1. Use hidden or minimized window? It seems there exists background throttling issues? https://github.com/tauri-apps/tauri/issues/5250
2. webworker? If I got it right, vscode uses web worker for browser native extensions. The main drawback is that node api is not supported, but node api is not a tauri stuff, so it is fine. However, I am wondering the performance and security aspects of web worker compared to multi process solution.
3. or any other possible solution?

**Comment by FabianLars** [comment, marked as answer]
I'd say that a hidden window is indeed the solution right now. Is background throttling even a concern for your usecase? It's mostly just about animations and setInterval.

No idea about webworkers, never used them.

as far as i know that's really all that tauri can offer, the WebView don't have any other feature we can use. The only thing left would be to bundle a nodejs runtime or something yourself with the app but i'd trust the browser sandbox more than the nodejs context tbh.

**Reply by yf-yang** [reply]
Got it, thank you for your kind reply 😄

**Reply by yf-yang** [reply]
A detailed reference of background throttling behavior: https://blog.chromium.org/2020/11/tab-throttling-and-more-performance.html
